File: src/main/java/Back/bailAutoComplet/BailAutoComplet/Controller/BailleurController.java
package Back.bailAutoComplet.BailAutoComplet.Controller;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import Back.bailAutoComplet.BailAutoComplet.Service.BailleurService;
import Back.bailAutoComplet.BailAutoComplet.model.Bailleur;


@RestController
@RequestMapping("/bailleur")
public class BailleurController {

    @Autowired
    private BailleurService bailleurService;

    @GetMapping
    public List<Bailleur> getAllBailleurs() {
        return bailleurService.getAllBailleurs();
    }

    @GetMapping("/{id}")
    public ResponseEntity<Bailleur> getBailleurById(@PathVariable Long id) {
        Bailleur bailleur = bailleurService.getBailleurById(id);
        if (bailleur != null) {
            return ResponseEntity.ok(bailleur);
        } else {
            return ResponseEntity.notFound().build();
        }
    }

    @PostMapping
    public ResponseEntity<Bailleur> createBailleur(@RequestBody Bailleur bailleur) {
        return ResponseEntity.ok(bailleurService.createBailleur(bailleur));
    }

    @PutMapping("/{id}")
    public ResponseEntity<Bailleur> updateBailleur(@PathVariable Long id, @RequestBody Bailleur bailleur) {
        return ResponseEntity.ok(bailleurService.updateBailleur(id, bailleur));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteBailleur(@PathVariable Long id) {
        bailleurService.deleteBailleur(id);
        return ResponseEntity.noContent().build();
    }
}

File: src/bailleur/bailleur.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  HttpCode,
  NotFoundException,
  Param,
  ParseIntPipe,
  Post,
  Put,
} from '@nestjs/common';
import { Bailleur } from './bailleur.entity';
import { BailleurService } from './bailleur.service';

@Controller('bailleur')
export class BailleurController {
  constructor(private readonly bailleurService: BailleurService) {}

  @Get()
  getAllBailleurs(): Promise<Bailleur[]> {
    return this.bailleurService.getAllBailleurs();
  }

  @Get(':id')
  async getBailleurById(
    @Param('id', ParseIntPipe) id: number,
  ): Promise<Bailleur> {
    const bailleur = await this.bailleurService.getBailleurById(id);

    if (!bailleur) {
      throw new NotFoundException();
    }

    return bailleur;
  }

  @Post()
  createBailleur(@Body() bailleur: Partial<Bailleur>): Promise<Bailleur> {
    return this.bailleurService.createBailleur(bailleur);
  }

  @Put(':id')
  updateBailleur(
    @Param('id', ParseIntPipe) id: number,
    @Body() bailleur: Partial<Bailleur>,
  ): Promise<Bailleur> {
    return this.bailleurService.updateBailleur(id, bailleur);
  }

  @Delete(':id')
  @HttpCode(204)
  deleteBailleur(@Param('id', ParseIntPipe) id: number): Promise<void> {
    return this.bailleurService.deleteBailleur(id);
  }
}
